### PR TITLE
EES-6211: New publications endpoint

### PR DESCRIPTION
**Pull request by leeoades**
This is part 2 of the piece of work to move the Search Function off of the soon-to-be-deprecated search API endpoint. The changes to the ContentAPI client now consume the newly introduced endpoints.

### CODE REVIEW

File: src/GovUk.Education.ExploreEducationStatistics.Content.Api/Controllers/PublicationController.cs [hunk 1]
```diff
@@ -17,98 +17,103 @@
...
+    public async Task<ActionResult<List<PublicationSitemapItemViewModel>>> ListSitemapItems(
+        CancellationToken cancellationToken = default) =>
+        await _publicationService.ListSitemapItems(cancellationToken)
+            .HandleFailuresOrOk();
+    
+    [HttpGet("publicationInfos")]
+    public async Task<ActionResult<IList<PublicationInfoViewModel>>> ListPublicationInfos(
+        [FromQuery] GetPublicationInfosRequest request, 
+        CancellationToken cancellationToken = default) =>
+        Ok(await _publicationService.ListPublicationInfos(request.ThemeId, cancellationToken));
```
Comment (benoutram) [inline src/GovUk.Education.ExploreEducationStatistics.Content.Api/Controllers/PublicationController.cs:118]: Looks like there are bunch of changes on this PR like this one and changes in QueryableExtensions, ArrayExtensions, GuidExtensions etc that already got merged to the dev branch by https://github.com/dfe-analytical-services/explore-education-statistics/pull/6036 so I'm unsure why they are showing in the diff of this PR. Could be because you reused the same branch?

File: src/GovUk.Education.ExploreEducationStatistics.Content.Services/Extensions/GuidExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿using System;
+using System.Diagnostics.CodeAnalysis;
+
+namespace GovUk.Education.ExploreEducationStatistics.Content.Services.Extensions;
+
+public static class GuidExtensions
+{
+    public static bool IsNotBlank([NotNullWhen(true)]this Guid? guid) =>
+        guid.HasValue && guid.Value != Guid.Empty;
```
[the remark this answers is not in the document]
Comment (leeoades) [reply src/GovUk.Education.ExploreEducationStatistics.Content.Services/Extensions/GuidExtensions.cs]: Actually, this is a very good example of me sometimes not following my own guidance! One should always create tests in the affirmative. So I have refactored this to be `IsBlank`. The reason I haven't called it `IsEmpty` is because `Empty` has a specific meaning for a Guid and we are also testing for `null`.

File: src/GovUk.Education.ExploreEducationStatistics.Content.Search.FunctionApp/Clients/ContentApi/ContentApiClient.cs [hunk 3]
```diff
@@ -62,162 +68,104 @@ public async Task<ReleaseSearchableDocument> GetPublicationLatestReleaseSearchab
...
+        var response = await Get<object>(GetThemesEndpoint, cancellationToken);
+        
         return response switch
         {
-            GetResponse<PaginatedResultDto<PublicationDto>>.Success => (true, null),
-            GetResponse<PaginatedResultDto<PublicationDto>>.Error error => (false, error.ErrorMessage),
+            GetResponse<object>.Success => (true, null),
+            GetResponse<object>.Error error => (false, error.ErrorMessage),
             _ => throw new UnreachableException()
         };
```
Comment (benoutram) [inline src/GovUk.Education.ExploreEducationStatistics.Content.Search.FunctionApp/Clients/ContentApi/ContentApiClient.cs:90]: Is it possible to use `Process` here like the rest of the methods, or is it because no exception gets thrown that we don't want to use it?

File: src/GovUk.Education.ExploreEducationStatistics.Common/Extensions/GuidExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,10 @@
+﻿using System;
+using System.Diagnostics.CodeAnalysis;
+
+namespace GovUk.Education.ExploreEducationStatistics.Common.Extensions;
+
+public static class GuidExtensions
+{
+    public static bool IsBlank([NotNullWhen(true)]this Guid? guid) =>
```
[the remark this answers is not in the document]
Comment (benoutram) [reply src/GovUk.Education.ExploreEducationStatistics.Common/Extensions/GuidExtensions.cs]: Cool, I understand now! Thanks.

**Review by benoutram — APPROVED** [review]
